pulsar-impl/src/main/java/com/datastax/oss/sink/pulsar/LocalSchemaRegistry.java

```diff
@@ -57,9 +57,10 @@ public PulsarSchema ensureAndUpdateSchema(String path, Object value) {
       if (res == null) {
         res = PulsarSchema.createFromStruct(path, struct, this);
         registry.put(path, res);
+      } else {
+        // need to recover nulls from previous records
+        res.update(path, struct, this);
```

Comment:
why do you need to move this ?

in the previous code we were always calling "res.update"
now we are calling it only when we see an already known field

it is a "optimization" or does is have a functional impact ?

if it is only a optimization, then I am not sure it is worth to change it, changing it may have side effects

Reply:
`PulsarSchema.createFromStruct` already calls it, so the at first pass the update is called twice with the same parameters and the method is idempotent

probably it's better to fix it in another pull

Reply:
https://github.com/datastax/pulsar-sink/blob/a81e620ccd5a9aa8aac40af48598857c04c99b6d/pulsar-impl/src/main/java/com/datastax/oss/sink/pulsar/PulsarSchema.java#L96-L100